File: Data Structures & Algorithms/maximum-ascending-subarray-sum/submission-0.py
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        maxSum = 0
        sum = 0
        prev = 0

        for num in nums:
            if num > prev:
                sum += num
            else:
                maxSum = max(maxSum, sum)
                sum = num
            prev = num
        
        maxSum = max(maxSum, sum)

        return maxSum

        

File: Data Structures & Algorithms/maximum-subarray/submission-0.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSoFar = -1000000000;
        int max = maxSoFar;

        for (int n : nums) {
            maxSoFar = Math.max(n, n + maxSoFar);
            max = Math.max(max, maxSoFar);
        }
        return max;
    }
}
